### PR TITLE
Add Russian translation

### PR DESCRIPTION
**Pull request by ZeusMods**
Hello! Thanks, for this cool project!
I decided to contribute and added support for Russian language.

If there are any problems, please write under pull request.

### CODE REVIEW

File: po/ru.po [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+msgstr ""
+"Project-Id-Version: spot\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2021-03-16 19:27+0100\n"
+"PO-Revision-Date: 2021-03-16 19:21+0100\n"
+"Last-Translator: Automatically generated\n"
+"Language-Team: Paul Bragin <zeusmods@protonmail.com>\n"
+"Language: ru\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=ASCII\n"
```
Comment (ondras12345) [inline po/ru.po:16]: Shouldn't this be UTF-8?
See https://raw.githubusercontent.com/GNOME/gimp/master/po/ru.po for an example of a Russian po file

File: po/ru.po [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=(n != 1);\n"
```
Comment (ondras12345) [inline po/ru.po:18]: According to [gettext documentation](https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html), Russian needs something like
```
Plural-Forms: nplurals=3; \
    plural=n%10==1 && n%100!=11 ? 0 : \
           n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;
```

File: po/ru.po (hunk 1 above)
Comment (ZeusMods) [reply po/ru.po:16]: I didn't really looked into the docs, so thx for fixing the stuff.

File: po/ru.po [hunk 3]
```diff
@@ -8,14 +8,15 @@ msgstr ""
...
 "Last-Translator: Automatically generated\n"
 "Language-Team: Paul Bragin <zeusmods@protonmail.com>\n"
 "Language: ru\n"
 "MIME-Version: 1.0\n"
-"Content-Type: text/plain; charset=ASCII\n"
+"Content-Type: text/plain; charset=UTF-8\n"
 "Content-Transfer-Encoding: 8bit\n"
-"Plural-Forms: nplurals=2; plural=(n != 1);\n"
+"Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n"
+"%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2);\n"
```
Comment (ondras12345) [inline po/ru.po:19]: Great, but you also need to update `msgstr[0]`, `msgstr[1]` and add `msgstr[2]` around line 100 to make use of the plural forms. I can't speak Russian, so I won't be able to help you with that.
All this line in the header does is that it specifies which plural form (which item from the msgstr array) should be used for different values of n (for example different numbers of selected songs).

File: po/ru.po [hunk 4]
```diff
@@ -8,14 +8,15 @@ msgstr ""
 "Project-Id-Version: spot\n"
 "Report-Msgid-Bugs-To: \n"
 "POT-Creation-Date: 2021-03-16 19:27+0100\n"
-"PO-Revision-Date: 2021-03-16 19:21+0100\n"
+"PO-Revision-Date: 2021-05-5 16:21+0100\n"
```
Comment (ondras12345) [inline po/ru.po:11]: ISO8601 requires that you write the date as 2021-05-05, not 2021-05-5

Comment (ZeusMods) [reply po/ru.po]: Should be resolved, now.

File: po/ru.po [hunk 5]
```diff
@@ -100,6 +100,7 @@ msgid "song selected"
 msgid_plural "songs selected"
 msgstr[0] "песня выбрана"
 msgstr[1] "песни выбраны"
+msgstr[1] "песни были выбраны"
```
Comment (ondras12345) [inline po/ru.po:103]: Shouldn't the added line be `msgstr[2]` instead of `msgstr[1]`?

Comment (ZeusMods) [reply po/ru.po:103]: Shit, I need give more attention. Fixed rn

**Comment by xou816** [comment]
Hi there, thanks for the contribution! :) and thanks @ondras12345 for chiming in and helping with the review :) just give me a heads up when this is good to be merged.

Oh and @ZeusMods, I am under the impression that you're editing these files by hand, which might be because of my README. I strongly recommend using a specialized GUI tool for this, it might help with things such as plural rules! :)

**Comment by ondras12345** [comment]
`ru` needs to be added to `po/LINGUAS`. Other than that, the format of the po file looks good. Again, I can't speak Russian...
Please note that I have **not tested** this in any way.
I would not call this a review, I just wanted to add a comment to a specific line in the changed file. I'm pretty sure the "add a single comment" button (not "start a review") did not cause my comment to be part of a review the last time I tried...

> If you can't build Spot locally, you may run the spot-snapshots action against your branch from your Github fork (...I think?) to obtain a flatpak debug build.

I don't think I have permission to do that, but @ZeusMods should.

**Comment by xou816** [comment]
Call it helpful comments if you prefer :p

But yes, once LINGUAS is updated we should be able to test with the snapshot build!

**Comment by ZeusMods** [comment]
Yes, I did edit the file by my hands. For me most importantly to have a
good and understandable localization. As I said earlier I didn't read the
docs about gettext.

ср, 5 мая 2021 г., 23:05 Alexandre Trendel ***@***.***>:

**Comment by xou816** [comment]
Brilliant, translations seem to work on my end (with the build downloaded from the github action). I'll merge tomorrow unless either of you finds something that needs fixing!

**Comment by ZeusMods** [comment]
> Brilliant, translations seem to work on my end (with the build downloaded from the github action). I'll merge tomorrow unless either of you finds something that needs fixing!

Could you send me a screenshot, or link to the build so I can see how it works?

**Comment by xou816** [comment]
Sure! You can download a build from the Checks tab at the top, then Artifacts

Capture d’écran du 2021-05-06 22-06-55

Capture d’écran du 2021-05-07 09-29-39

**Comment by ZeusMods** [comment]
So i decided to fix some flaws in my translations, and also found a program that's works with .po file format.
https://poedit.net/
I will update my localization soon.

**Comment by ZeusMods** [comment]
Offtopic: Wanna a funny thing? The program doesn't see my language, damn it. So i am unable to test how it looks. But it should be good tho.

**Comment by xou816** [comment]
Something approaching yes!

`flatpak run --env=LANG=ru_RU.UTF-8 dev.alextren.Spot`

Also, depending on your setup, you might have to reconfigure the GNOME runtime to use Russian as well (was my case but yours might be already configured):

`flatpak update --subpath=ru org.gnome.Platform.Locale`

**Comment by ZeusMods** [comment]
Hooray! It works!

**Comment by ZeusMods** [comment]
I like my localization it works very well. I am satisfied. The localization is ready to merge!

**Comment by xou816** [comment]
Great, thanks for the contribution! :)

**Pull request merged**